src/shogun/lib/SGMatrix.h

```diff
@@ -258,6 +301,13 @@ template<class T> class SGMatrix : public SGReferencedData
 		/** Returns a const iterator to the element following the last element of the container. */
 		const_iterator end() const noexcept { return const_iterator(matrix + (num_rows * num_cols)); }
 
+		/** Returns an iterator to the first element of the container. */
+		column_iterator begin_column() noexcept { return column_iterator(matrix); }
+
+		/** Returns an iterator to the element following the last column of the container. */
+		column_iterator end_column() noexcept { return column_iterator(matrix + (num_rows * num_cols)); }
```

Comment:
definitely not...

Reply:
Hmm 
```column_iterator end_column() noexcept{ return column_iterator(*this,num_cols); }``` ?

Reply:
Yes

Reply:
.begin_column() and .end_column() are working perfectly fine !.  There's a small build error during for_each comparison. Need to fix that and we're done.